extras-scalapb/src/main/scala/com/avast/clients/rabbitmq/extras/format/ScalaPBAsJsonProductConverter.scala

```diff
@@ -0,0 +1,42 @@
+package com.avast.clients.rabbitmq.extras.format
+
+import com.avast.bytes.Bytes
+import com.avast.clients.rabbitmq.ProductConverter
+import com.avast.clients.rabbitmq.api.{ConversionException, MessageProperties}
+import scalapb.GeneratedMessage
+import scalapb.json4s.Printer
+
+import scala.annotation.implicitNotFound
+import scala.reflect.ClassTag
+import scala.util.control.NonFatal
+
+@implicitNotFound("Could not generate ScalaPBAsJsonProductConverter for ${A}, try to import or define some")
+trait ScalaPBAsJsonProductConverter[A <: GeneratedMessage] extends ProductConverter[A]
+
+object ScalaPBAsJsonProductConverter {
+  def derive[A <: GeneratedMessage: ScalaPBAsJsonProductConverter](): ScalaPBAsJsonProductConverter[A] =
+    implicitly[ScalaPBAsJsonProductConverter[A]]
```

Comment:
will this work? wouldn't `implicitly` just loop back to `derive` defined on the line above?

Reply:
This is the same mechanism as used in existing module and it works there: https://github.com/avast/rabbitmq-scala-client/blob/master/extras-circe/src/main/scala/com/avast/clients/rabbitmq/extras/format/JsonProductConverter.scala#L18
Maybe @jendakol could comment it.

Reply:
oh, my bad! (my comment above would be valid concern, only if `def derive` was itself `implicit`)

that being said, this is usually called `apply` (not `derive`) in every other library, but it's probably better to keep with the custom that's being used in `rabbitmq-scala-client`